src/core/PathUtil.cpp

```diff
@@ -31,6 +34,22 @@ namespace PathUtil
 			case Base::DefaultSoundfont : loc = ConfigManager::inst()->userSf2Dir(); break;
 			case Base::UserGIG          : loc = ConfigManager::inst()->gigDir(); break;
 			case Base::DefaultGIG       : loc = ConfigManager::inst()->userGigDir(); break;
+			case Base::LocalDir    :
+			{
+				Song* s = Engine::getSong();
+				if (s)
+				{
+					QString projectPath = s->projectFileName();
+					loc = QFileInfo(projectPath).path();
+				}
+				else
+				{
+					// Keeps the base prefix so the method calling this
+					// can handle the situation.
+					return QString("local:");
```

Comment:
This feels like a hack. The entire point of this function is to return the location of a base, and yet this returns a prefix. I think this should probably be handled by the caller.

Also, if you did want this here it should return basePrefix(Base::LocalDir) so the prefixes are only defined once.

Reply:
The caller handles it, but to know that it will have to handle it we need to keep the prefix. An empty string would result in a valid path, so keeping the prefix looked like a good alternative to signal the caller that the path wasn't resolved. This has to be done because at a very particular case (when creating a bundle from the CLI) the `PathUtil` doesn't have access to a `Song` object because it isn't created (we are writing the bundle straight from the `DataFile`).

It's a good idea change it to `basePrefix(Base::LocalDir)` though, because then we don't need to replace it if we change the prefix in the future!

Reply:
Done! Using `basePrefix` instead now